scripts/get_deployed_brackets.js

```diff
@@ -0,0 +1,65 @@
+const createCsvWriter = require("csv-writer").createObjectCsvWriter
+const exchangeUtils = require("@gnosis.pm/dex-contracts")
+const { fetchTokenInfoFromExchange, getExchange, getDeployedBrackets } = require("./utils/trading_strategy_helpers")(
+  web3,
+  artifacts
+)
+
+const argv = require("./utils/default_yargs")
+  .option("masterSafe", {
+    type: "string",
+    describe: "Address of Gnosis Safe owning every bracket",
+    demandOption: true,
+  })
+  .option("outputFile", {
+    type: "string",
+    describe: "Path and file name for the output of the script",
+    demandOption: false,
+  }).argv
+
+module.exports = async (callback) => {
+  try {
+    const bracketAddresses = await getDeployedBrackets(argv.masterSafe)
+
+    // writing the brackets into a csv file
+    const csvWriter = createCsvWriter({
+      path: argv.outputFile || "./bracket-addresses.csv",
+      header: [
+        { id: "Address", title: "Address" },
+        { id: "Type", title: "Type" },
+        { id: "Description", title: "Description" },
+        { id: "Tags", title: "Tags" },
+        { id: "EL", title: "Etherscan link" },
+        { id: "CP", title: "Easy C&P for JSON/SQL" },
+      ],
+    })
+
+    const exchange = await getExchange(web3)
+    const records = await Promise.all(
+      bracketAddresses.map(async (bracketAddress) => {
+        const orders = exchangeUtils.decodeOrdersBN(await exchange.getEncodedUserOrders.call(bracketAddress))
+        let tradingPair
+        if (orders.length > 0) {
+          const tokenInfoPromises = fetchTokenInfoFromExchange(exchange, [orders[0].buyToken, orders[0].sellToken])
+          tradingPair =
+            (await tokenInfoPromises[orders[0].sellToken]).symbol + " - " + (await tokenInfoPromises[orders[0].buyToken]).symbol
```

Comment:
A bit of replication here. Maybe you can simplify, and wait just once (Promise.all). Also, destructuring the array/objects may come handy, together with using
Good idea to add auxiliary consts, like `const firstOrder` or  `tokenInfo`.

Reply:
we don't have any repetition, as the promises are stored in a kinda global object 
`  
const globalTokenPromisesFromId = {}
`

Reply:
With repetition I mean you can se `orders[0]` many times,
`orders[0].buyToken`

Then it's too complex construction of the string, it's a good practice to aim to do one thing in each line (for readability), here you are concatenating, so better not to await two promise, get the first order (2 times) and extract the symbol from the resulting object too

No shame in having more lines and aux const/vars:
```js
const { sellToken, buyOrder } = orders[0]
[ sellTokenInfo, buyTokenInfo ] = await Promise.all([
   tokenInfoPromises[sellToken]),
   tokenInfoPromises[buyToken]) 
])
tradingPair = `${sellTokenInfo.symbol}-${buyTokenInfo.symbol}`
```

Each line above have a single purpose, it's not bad in my view even to have:
```
const firstOrder = orders[0]
const { sellToken, buyOrder } = firstOrder
```
But also, it's fair to do it all at once